File: src/2025-03-04-03-28-40--deepseek-chat-0304/javascript/exercises/practice/food-chain/food-chain.js
export class Song {
  constructor() {
    this.animals = [
      { name: 'fly', comment: "I don't know why she swallowed the fly. Perhaps she'll die." },
      { name: 'spider', action: 'wriggled and jiggled and tickled inside her', comment: 'She swallowed the spider to catch the fly.' },
      { name: 'bird', exclamation: 'How absurd to swallow a bird!', comment: 'She swallowed the bird to catch the spider that wriggled and jiggled and tickled inside her.' },
      { name: 'cat', exclamation: 'Imagine that, to swallow a cat!', comment: 'She swallowed the cat to catch the bird.' },
      { name: 'dog', exclamation: 'What a hog, to swallow a dog!', comment: 'She swallowed the dog to catch the cat.' },
      { name: 'goat', exclamation: 'Just opened her throat and swallowed a goat!', comment: 'She swallowed the goat to catch the dog.' },
      { name: 'cow', exclamation: "I don't know how she swallowed a cow!", comment: 'She swallowed the cow to catch the goat.' },
      { name: 'horse', exclamation: "She's dead, of course!" }
    ];
  }

  verse(n) {
    const animal = this.animals[n - 1];
    let lines = [`I know an old lady who swallowed a ${animal.name}.`];
    
    if (animal.exclamation) {
      lines.push(animal.exclamation);
    }
    
    if (animal.action) {
      lines.push(`It ${animal.action}.`);
    }
    
    if (n === 8) {
      return lines.join('\n') + '\n';
    }
    
    for (let i = n - 1; i >= 0; i--) {
      if (this.animals[i].comment) {
        lines.push(this.animals[i].comment);
      }
    }
    
    return lines.join('\n') + '\n';
  }

  verses(start, end) {
    let result = [];
    for (let i = start; i <= end; i++) {
      result.push(this.verse(i));
    }
    return result.join('\n') + '\n';
  }
}

File: src/2025-03-04-03-28-40--deepseek-chat-0304/go/exercises/practice/food-chain/food_chain.go
package foodchain

import (
	"fmt"
	"strings"
)

var animals = []string{
	"fly",
	"spider",
	"bird",
	"cat",
	"dog",
	"goat",
	"cow",
	"horse",
}

var descriptions = []string{
	"",
	"It wriggled and jiggled and tickled inside her.",
	"How absurd to swallow a bird!",
	"Imagine that, to swallow a cat!",
	"What a hog, to swallow a dog!",
	"Just opened her throat and swallowed a goat!",
	"I don't know how she swallowed a cow!",
	"She's dead, of course!",
}

func Verse(v int) string {
	if v < 1 || v > 8 {
		return ""
	}

	verse := fmt.Sprintf("I know an old lady who swallowed a %s.\n", animals[v-1])
	
	if v == 8 {
		return verse + descriptions[v-1]
	}

	if descriptions[v-1] != "" {
		verse += descriptions[v-1] + "\n"
	}

	for i := v - 1; i > 0; i-- {
		verse += fmt.Sprintf("She swallowed the %s to catch the %s", animals[i], animals[i-1])
		if i == 2 {
			verse += " that wriggled and jiggled and tickled inside her"
		}
		verse += ".\n"
	}

	verse += "I don't know why she swallowed the fly. Perhaps she'll die."
	return verse
}

func Verses(start, end int) string {
	var verses []string
	for i := start; i <= end; i++ {
		verses = append(verses, Verse(i))
	}
	return strings.Join(verses, "\n\n")
}

func Song() string {
	return Verses(1, 8)
}
